packages/api/src/hooks/useAccountsList.ts

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from 'react';
+import useAuthorize from './useAuthorize';
+
+/** A custom hook that returns the list of accounts of the logged in user. */
+const useAccountsList = () => {
+    const { data: authorize_data, ...rest } = useAuthorize();
+
+    // Add additional information to the authorize response.
+    const modified_accounts = useMemo(() => {
+        return authorize_data.account_list?.map(account => {
+            return {
+                ...account,
+                /** Creation time of the account. */
+                created_at: account.created_at && new Date(account.created_at),
+                /** Date till client has excluded him/herself from the website, only present if client is self excluded. */
+                excluded_until: account.excluded_until && new Date(account.excluded_until),
```

Comment:
@thisyahlen-deriv Revert this check to ternary, `&&` will make the type to also be a `boolean`